### PR TITLE
Add annotation and label for MachineDeployment details

### PR DESCRIPTION
**Pull request by tuommaki**
Node Pool ID and allocated subnet are marked on MachineDeployment object as
label and annotation correspondingly.

### CODE REVIEW

File: kubernetes/annotations_and_labels.md [hunk 1]
```diff
@@ -8,6 +8,9 @@ This page defines common annotations and labels we set in Kubernetes objects.
   location. For now this should be in our `giantswarm/giantswarm` repository in
   which we crosslink all pages to create a reasonable documentation of all kinds
   of Kubernetes resources, their functionality and relationships.
+- `machine-deployment.giantswarm.io/subnet` - value should contain a subnet
```
Comment (tuommaki) [inline kubernetes/annotations_and_labels.md]: This is managed by `aws-operator`.

File: kubernetes/annotations_and_labels.md [hunk 2]
```diff
@@ -20,6 +23,9 @@ This page defines common annotations and labels we set in Kubernetes objects.
   Secrets and CertConfigs.
 - `giantswarm.io/cluster` - value should contain tenant cluster ID which this
   object is part of. E.g. `giantswarm.io/cluster=eggs2`.
+- `giantswarm.io/machine-deployment` - value should contain tenant cluster node
```
Comment (tuommaki) [inline kubernetes/annotations_and_labels.md]: This is managed by `api` atm..

File: kubernetes/annotations_and_labels.md (hunk 1 above)
Comment (xh3b4sd) [reply kubernetes/annotations_and_labels.md]: Just for the record, the description describes the desired usage. The current implementation tracks the whole cluster CIDR for transitioning reasons.

Comment (tuommaki) [reply kubernetes/annotations_and_labels.md]: True, also noted that this is not a place to use `should`... :roll_eyes:

File: kubernetes/annotations_and_labels.md [hunk 3]
```diff
@@ -8,6 +8,9 @@ This page defines common annotations and labels we set in Kubernetes objects.
   location. For now this should be in our `giantswarm/giantswarm` repository in
   which we crosslink all pages to create a reasonable documentation of all kinds
   of Kubernetes resources, their functionality and relationships.
+- `machine-deployment.giantswarm.io/subnet` - value contains a subnet CIDR that
+  has been allocated for the given node pool / MachineDeployment object that
+  has this annotation.
```
Comment (xh3b4sd) [inline kubernetes/annotations_and_labels.md]: Most other descriptions have an example. That would be cool here as well. Then we see the IP and Mask.

**Review by xh3b4sd — APPROVED** [review]

File: kubernetes/annotations_and_labels.md (hunk 2 above)
Comment (MarcelMue) [reply kubernetes/annotations_and_labels.md]: All this should sounds weird but we have it in other descriptions already.

**Review by MarcelMue — APPROVED** [review]
fine with me

File: kubernetes/annotations_and_labels.md [hunk 4]
```diff
@@ -8,6 +8,9 @@ This page defines common annotations and labels we set in Kubernetes objects.
   location. For now this should be in our `giantswarm/giantswarm` repository in
   which we crosslink all pages to create a reasonable documentation of all kinds
   of Kubernetes resources, their functionality and relationships.
```
Comment (kopiczko) [inline kubernetes/annotations_and_labels.md]: It should be specified where we set those annotations/labels. See https://github.com/giantswarm/fmt/blob/31541cd695647a0461069f0337f4fd3d3485e5c6/kubernetes/annotations_and_labels.md#tenant-cluster

**Review by kopiczko — CHANGES_REQUESTED** [review]
See my comment.

File: kubernetes/annotations_and_labels.md [hunk 5]
```diff
@@ -56,6 +63,25 @@ This page defines common annotations and labels we set in Kubernetes objects.
...
+
+##### Annotations
+
+##### Labels
+
+#### MachineDeployment
+
+##### Annotations
+
+- `machine-deployment.giantswarm.io/subnet`
```
Comment (tuommaki) [inline kubernetes/annotations_and_labels.md]: @kopiczko Like this?

**Review by kopiczko — APPROVED** [review]

**Review by rossf7 — APPROVED** [review]

**Pull request merged**